cellpack/autopack/Environment.py

```diff
@@ -280,6 +280,23 @@ def _setup(self):
                 self.referenced_objects,
             ) = Recipe.resolve_composition(self.recipe_data)
             self.create_objects()
+        if self.use_gradient:
+            gradients = self.recipe_data["gradients"]
+            for gradient_name in gradients:
+                gradient_data = gradients[gradient_name]
+                if "surface_name" in gradient_data:
+                    gradient_data["object"] = self.get_compartment_object_by_name(
+                        gradient_data["surface_name"]
+                    )
+                self.set_gradient(gradient_name, gradient_data)
+
+    def get_compartment_object_by_name(self, compartment_name):
+        """
+        Returns compartment object by name
+        """
+        for compartment in self.compartments:
+            if compartment.name == compartment_name:
+                return compartment
```

Comment:
This is essentially reimplementing a dictionary lookup as a linear search.... I know we never have that many compartments but maybe we will some day? How plausible is it to just make `self.compartments` a dictionary? If you need the list you could just `self.compartments.values` etc.

Reply:
at the moment we're mostly always iterating over all compartments so I think it makes sense to have it be an array